crates/biome_cli/src/execute/mod.rs

```diff
@@ -532,12 +536,15 @@ pub fn execute_mode(
     project_key: ProjectKey,
 ) -> Result<(), CliDiagnostic> {
     // If a custom reporter was provided, let's lift the limit so users can see all of them
-    execution.max_diagnostics = if cli_options.reporter.is_default() {
+    execution.max_diagnostics = if cli_options
+        .reporter
+        .iter()
+        .any(|reporter| reporter.is_default())
+    {
         cli_options.max_diagnostics.into()
     } else {
         info!(
-            "Removing the limit of --max-diagnostics, because of a reporter different from the default one: {}",
-            cli_options.reporter
+            "Removing the limit of --max-diagnostics, because reporter(s) does not include the default one"
         );
         u32::MAX
```

Comment:
Unfortunately, this logic must change. With multiple reporters, we can pass the default reporter plus others. The limit must still be applied to the terminal (default) reporter.